.github/workflows/go.yml

```
@@ -1,23 +1,43 @@
-name: i386 linux tests
+name: Run unit tests on Linux
 
 on:
   push:
-    branches: [ master ]
+    branches: [ main ]
   pull_request:
-    branches: [ master ]
+    branches: [ main ]
   workflow_dispatch:
 
+env:
+  CGO_CFLAGS_ALLOW: "-O -D__BLST_PORTABLE__"
+  CGO_CFLAGS: "-O -D__BLST_PORTABLE__"
+
 jobs:
   build:
-    runs-on: self-hosted
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [ '1.20', '1.21.x' ]
+        goarch: ["386", amd64, arm64]
+
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up Go
-        uses: actions/setup-go@v2
+      - uses: actions/checkout@v4
+      - name: Setup Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v4
         with:
-          go-version: 1.21.4
-      - name: Run tests
-        run: go test -short ./...
-        env:
-          GOOS: linux
-          GOARCH: 386
+          go-version: ${{ matrix.go-version }}
+          # You can test your matrix by printing the current Go version
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+      # - name: Lint contracts
+      #   run: cd suave && forge fmt --check
+      # - name: Build contracts
+      #   run: cd suave && forge build
+      - name: Display Go version
+        run: go version
+      - name: Run unit tests
+        run: go test -short ./accounts ./cmd/geth ./core ./core/types ./core/vm ./eth/... ./internal/ethapi/... ./miner ./params #./suave/...
```

do you still need this?
```suggestion
        run: go test -short ./accounts ./cmd/geth ./core ./core/types ./core/vm ./eth/... ./internal/ethapi/... ./miner ./params
```